pkgs/macro_service/lib/src/macro_service.g.dart

```diff
@@ -40,6 +40,32 @@ extension type HostEndpoint.fromJson(Map<String, Object?> node) {
   int get port => node['port'] as int;
 }
 
+enum HostRequestType {
+  unknown,
+  augmentRequest;
+}
+
+extension type HostRequest.fromJson(Map<String, Object?> node) {
+  static HostRequest augmentRequest(AugmentRequest augmentRequest) =>
+      HostRequest.fromJson(
+          {'type': 'AugmentRequest', 'value': augmentRequest.node});
+  HostRequestType get type {
+    switch (node['type'] as String) {
+      case 'AugmentRequest':
```

Comment:
nit: We should consider serializing these using the enum value instead of the String value.

Reply:
Quite possibly, what are the benefits of using the enum value?

I think I liked the name with initial upper case letter more as it matches the class name, i.e. it seems to make more sense that we refer to `AugmentRequest` on the wire than `augmentRequest`, which is what we'd get from the enum value.

Other than that I don't remember any particularly strong reason for it being like it is now :)

Thanks.

Reply:
Just because an integer is a lot smaller

Reply:
In general maybe this is a fallout of using json schema... but we are talking about significantly larger payloads it seems like in general if we can't optimize things like this.

Maybe, in the binary form we can actually serialize these differently and use the index, and in the regular JSON form we don't?

Reply:
Ah, got it.

As far as the schema goes I think we definitely want the string, we don't want to mess with enum ordering breaking the protocol :)

Re: the binary format, optimizing for strings is anyway a concern, pretty sure we'll end up encoding references to a repeated string as an ID, it's easy enough to do and it saves doing the UTF8 conversion lots of times. So that should cover this case, too.